spark/interpreter/src/main/scala/org/apache/zeppelin/spark/SparkZeppelinContext.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.spark
+
+import java.util
+
+import org.apache.spark.SparkContext
+import org.apache.zeppelin.annotation.ZeppelinApi
+import org.apache.zeppelin.display.AngularObjectWatcher
+import org.apache.zeppelin.display.ui.OptionInput.ParamOption
+import org.apache.zeppelin.interpreter.{BaseZeppelinContext, InterpreterContext, InterpreterHookRegistry}
+
+import scala.collection.{JavaConversions, Seq}
+
+
+/**
+  * ZeppelinContext for Spark
+  */
+class SparkZeppelinContext(val sc: SparkContext,
+                           val sparkShims: SparkShims,
+                           val hooks2: InterpreterHookRegistry,
+                           val maxResult2: Int) extends BaseZeppelinContext(hooks2, maxResult2) {
+
+  private val interpreterClassMap = Map(
+    ("spark", "org.apache.zeppelin.spark.SparkInterpreter"),
+    ("sql", "org.apache.zeppelin.spark.SparkSqlInterpreter"),
+    ("dep", "org.apache.zeppelin.spark.DepInterpreter"),
+    ("pyspark", "org.apache.zeppelin.spark.PySparkInterpreter")
+  )
+
+  private val supportedClasses = scala.collection.mutable.ArrayBuffer[Class[_]]()
+
+  try
+    supportedClasses += Class.forName("org.apache.spark.sql.Dataset")
+  catch {
+    case e: ClassNotFoundException =>
+  }
+
+  try
+    supportedClasses += Class.forName("org.apache.spark.sql.DataFrame")
+  catch {
+    case e: ClassNotFoundException =>
+
+  }
+  if (supportedClasses.isEmpty) throw new RuntimeException("Can not load Dataset/DataFrame class")
+
```

Comment:
Since both the supported classes are part of the Scala SDK, should it be safe to just define it like this, and avoid the `forName` calls:

 ```
 val supportedClasses: scala.collection.mutable.Seq[Class[_]] = scala.collection.mutable.Seq(
      classOf[org.apache.spark.sql.Dataset[_]], classOf[org.apache.spark.sql.DataFrame])

```

Reply:
I am afraid not, because zeppelin needs to support both spark 1.x and spark 2.x. And Dataset is not available in spark 1.x